Common++/header/Logger.h

```diff
@@ -113,10 +107,124 @@ namespace pcpp
+		constexpr LogSource() = default;
+
+		/**
+		 * Constructor for LogSource with only the log module.
+		 * @param logModule The log module.
+		 */
+		explicit constexpr LogSource(LogModule logModule) : logModule(logModule)
+		{}
+
+		/**
+		 * Constructor for LogSource with all parameters.
+		 * @param logModule The log module.
+		 * @param file The source file.
+		 * @param function The source function.
+		 * @param line The line number.
+		 */
+		constexpr LogSource(LogModule logModule, const char* file, const char* function, int line)
+		    : file(file), function(function), line(line), logModule(logModule)
+		{}
+
+		const char* file = nullptr;               /**< The source file. */
+		const char* function = nullptr;           /**< The source function. */
+		int line = 0;                             /**< The line number. */
+		LogModule logModule = UndefinedLogModule; /**< The log module. */
+	};
+
+	/**
+	 * An enum representing the log level. Currently 4 log levels are supported: Off, Error, Info and Debug. Info is the
+	 * default log level
+	 */
+	enum class LogLevel
+	{
+		Off = PCPP_LOG_LEVEL_OFF,      ///< No log messages are emitted.
+		Error = PCPP_LOG_LEVEL_ERROR,  ///< Error level logs are emitted.
+		Info = PCPP_LOG_LEVEL_INFO,    ///< Info level logs and above are emitted.
+		Debug = PCPP_LOG_LEVEL_DEBUG   ///< Debug level logs and above are emitted.
+	};
+
+	inline std::ostream& operator<<(std::ostream& s, LogLevel v)
+	{
+		return s << static_cast<std::underlying_type<LogLevel>::type>(v);
+	}
+
+	// Forward declaration
+	class Logger;
+
+	namespace internal
+	{
+		/**
+		 * @class LogContext
+		 * @brief A context encapsulating the details of a single log message to be passed to the Logger.
+		 */
+		class LogContext
+		{
+		public:
+			friend class pcpp::Logger;
+
+			/**
+			 * @brief Creates a context with an empty message with Info level and no source.
+			 */
+			LogContext() = default;
+			/**
+			 * @brief Creates a context with an empty message with the given level and source.
+			 * @param level The log level for this message.
+			 * @param source The log source.
+			 */
+			explicit LogContext(LogLevel level, LogSource const& source = {}) : level(level), m_Source(source)
+			{}
+
+			/**
+			 * @brief Initializes the context with an empty message and the given level and source.
+			 * @param level The log level for this message.
+			 * @param source The log source.
+			 */
+			void init(LogLevel level, LogSource const& source)
+			{
+				m_Source = source;
+				this->level = level;
+				m_Stream.clear();
+				m_Stream.str({});
+			}
+
+			/**
+			 * @brief Appends to the message.
+			 * @param value The value to append.
+			 * @return A reference to this context.
+			 */
+			template <class T> inline LogContext& operator<<(T const& value)
+			{
+				m_Stream << value;
+				return *this;
+			}
+
+			/**
+			 * @brief The log level at which the message will be emitted.
+			 */
+			LogLevel level = LogLevel::Info;
```

Comment:
It is left from prior to `createLogContext` having overloads to accept `LogSource` and `LogLevel`, so the level needed to be switched manually afterwards. I suppose it can be set to private, as it is not used directly atm.

Reply:
Fixed e972e67